### PR TITLE
Tests for listings

### LINKED ISSUE #6: Add integration tests related to Listings (GET)

### PR DESCRIPTION
**Pull request by wwajerowicz**
Fixes #6

### CODE REVIEW

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala [hunk 1]
```diff
@@ -42,44 +46,51 @@ class AdminSpec extends BaseSpec with OptionValues with CancelAfterFailure with
...
-          val permissions = json.hcursor.get[Vector[Json]]("acl").toOption.value.foldLeft(Set.empty[String]) {
+      cl(Req(uri = s"$iamBase/acls/*?parents=true", headers = headersGroup)).mapJson { (json, result) =>
+        result.status shouldEqual StatusCodes.OK
+        val permissions =
+          json.hcursor.get[Vector[Json]]("acl").toOption.value.foldLeft(Set.empty[(String, Set[String])]) {
             case (acc, j) =>
               j.hcursor.downField("identity").get[String]("sub") match {
                 case Right(s) if s == config.iam.userSub =>
-                  j.hcursor.get[Set[String]]("permissions").toOption.value ++ acc
+                  acc + Tuple2(j.hcursor.get[String]("path").toOption.value,
```
Comment (umbreak) [inline src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: Is the Tuple2 needed? shouldn't `a -> b` or (a, b) should be enough?

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala [hunk 2]
```diff
@@ -576,5 +592,154 @@ class AdminSpec extends BaseSpec with OptionValues with CancelAfterFailure with
...
+              "_id" -> Json.fromString(s"$adminBase/projects/$id"),
+              "_source" -> Json.obj(
+                "@id" -> Json.fromString(s"$adminBase/projects/$id")
+              )
+            )
+          }: _*
+        )
+      }
+
+      def projectListingResultsWithAllFields(ids: Seq[String]): Json = {
```
Comment (umbreak) [inline src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: Can't you use a json template (json file with replacements) as we did in other places. In my opinion that increases readability

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala [hunk 3]
```diff
@@ -576,5 +592,154 @@ class AdminSpec extends BaseSpec with OptionValues with CancelAfterFailure with
...
+        }
+      }
+
+      "return only projects the user has access to" in {
+        val expectedResults = Json.obj(
+          "@context" -> Json.fromString(config.prefixes.searchContext.toString),
+          "_results" -> projectListingResults(projectIds),
+          "_total"   -> Json.fromInt(projectIds.size),
+          "_links" -> Json.obj(
+            "_self" -> Json.fromString(s"$adminBase/projects")
```
Comment (umbreak) [inline src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: Why is this self still here? I thought we removed this predicate

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala [hunk 4]
```diff
@@ -576,5 +592,154 @@ class AdminSpec extends BaseSpec with OptionValues with CancelAfterFailure with
...
+        }
+      }
+
+      "return only projects the user has access to with all fields" in {
+        val expectedResults = Json.obj(
+          "@context" -> Json.fromString(config.prefixes.searchContext.toString),
+          "_results" -> projectListingResultsWithAllFields(projectIds),
+          "_total"   -> Json.fromInt(projectIds.size),
+          "_links" -> Json.obj(
+            "_self" -> Json.fromString(s"$adminBase/projects?fields=nxv:_all")
```
Comment (umbreak) [inline src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: same here

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala (hunk 2 above)
Comment (wwajerowicz) [reply src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: will do

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala (hunk 3 above)
Comment (wwajerowicz) [reply src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: I think we agreed to remove it from source and single resource view, but still leave it in search results?

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala (hunk 1 above)
Comment (wwajerowicz) [reply src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: for some reason scala compiler didn't like `(a, b)`, but `(a -> b)` seems to work, will change that.

File: src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala (hunk 3 above)
Comment (umbreak) [reply src/test/scala/ch/epfl/bluebrain/nexus/tests/admin/AdminSpec.scala]: From my point of view it adds no value anywhere

**Pull request merged**